### PR TITLE
CellEditor - Iteration I

### PR DESCRIPTION
**Pull request by oliver----**

### CODE REVIEW

**Comment by oliver----** [comment]
Shall I review?

**Comment by michael** [comment]
Too early I would say... just tinkering around to get aware of all pieces involved, and set up new architecture and workflow.

**Comment by oliver----** [comment]
@michael The slow response after confirming a cell, was related to the hierarchical listener. I have disabled this in substance. Let's see when we really need this, and then address it in the right way.

**Comment by oliver----** [comment]
We should do some tidy-up and then get this merged.

File: src/sheet/SheetEditor.js [hunk 1]
```diff
@@ -350,8 +350,25 @@ export default class SheetEditor extends AbstractEditor {
...
     let cell = this._getAnchorCell()
     let newValue = this._formulaEditorContext.node.getText()
+    let { anchorRow, anchorCol} = this._getSheetSelection()
+
+    // Select cell below
+    anchorRow += 1
+
     editorSession.transaction(tx => {
       tx.set(cell.getPath(), newValue)
+      tx.setSelection({
```
Comment (oliver----) [inline src/sheet/SheetEditor.js]: There is a method in SheetComponent, which also shifts the viewport:
```
this.refs.sheet._nav(1, 0)
```
On the long run we should tidy-up the API of SheetComponent

**Comment by oliver----** [comment]
@michael it would be good to rebase onto develop, resolving the merge conflict

**Comment by michael** [comment]
I'm doing that right now.

File: src/sheet/SheetEditor.js (hunk 1 above)
Comment (michael) [reply src/sheet/SheetEditor.js]: Wanted to do it one flow. I guess we should have the _nav as a utility returning a new selection, and handle the this.send stuff in a separate method.

Comment (oliver----) [reply src/sheet/SheetEditor.js]: I see. Yeah, I need to think how to make the SheetComponent API useful.
So let's stick with your implementation, and fix SheetComponent in a different iteration

File: src/sheet/SheetEditor.js [hunk 2]
```diff
@@ -350,25 +350,10 @@ export default class SheetEditor extends AbstractEditor {
...
-          type: 'range',
-          anchorRow,
-          anchorCol,
-          focusRow: anchorRow,
-          focusCol: anchorCol
-        },
-        surfaceId: this.refs.sheet.getId()
-      })
+      let newSel = this.refs.sheet.shiftSelection(1, 0, false)
+      tx.setSelection(newSel)
```
Comment (michael) [inline src/sheet/SheetEditor.js]: @oliver---- better like that?

Comment (oliver----) [reply src/sheet/SheetEditor.js]: Yeah. Better that we don't see the internal selection data.
And with re viewport: SheetComponent should be smart enough to update the viewport automatically
If it doesn't we will fix it later.

**Comment by oliver----** [comment]
I think it is good for this iteration.

**Comment by oliver----** [comment]
Sweet.

**Comment by oliver----** [comment]
Let's call it a PR. Want to push da button?

**Comment by oliver----** [comment]
Ersta :)

**Comment by michael** [comment]
@oliver---- I don't understand how this code works 🤷‍♂️

https://github.com/stencila/stencila/blob/a38239ab900a9a17cde36b69e76826304e2309c7/src/sheet/SheetComponent.js#L383-L399

**Comment by michael** [comment]
like an anonymous block thing...

**Comment by oliver----** [comment]
Jui. Yeah that's ugly. Probably I have moved this from somewhere else and 'fixed' it using the block.
But honestly, a bit embarrassing ;) need to fix. Thx for the TODO

> Note: it is working because newFocusRow/Col has been determined above that block.

**Comment by oliver----** [comment]
@michael This was good work! Thx

**Pull request merged**